front/components/agent_builder/capabilities/mcp/MCPServerViewsSheet.tsx

```diff
@@ -283,11 +284,12 @@ export function MCPServerViewsSheet({
       setSelectedToolsInSheet([]);
 
       const action = mode.action;
-      if (
-        isMCPActionWithConfiguration(action) &&
-        allMcpServerViews.length > 0
-      ) {
-        const mcpServerView = allMcpServerViews.find(
+      if (isMCPActionWithConfiguration(action)) {
+        const labeledViews: MCPServerViewTypeWithLabel[] = [
```

Comment:
Just wondering why we don't set the value of mcpServerViews to mcpServerViewsWithLabel directly in the context provider 🤷🏻‍♂️

Reply:
Yeah that was my first intention but it has quite a big cascading effect and would require to change endpoints type and so on..